segfault compiling openssh when building sshnpd-slim Docker image

### Describe the bug

The dockerhub_sshnpd workflow is failing on the sshnpd-slim job (e.g. [this run](https://github.com/atsign-foundation/noports/actions/runs/12947257343/job/36114741559)):

```log
550.6 gcc  -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include  -DBSAES_ASM -DECP_NISTZ256_ASM -DECP_SM2P256_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_MONT -DOPENSSL_CPUID_OBJ -DOPENSSL_SM3_ASM -DPOLY1305_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DSM4_ASM -DVPAES_ASM -DVPSM4_ASM -fPIC -pthread -Wa,--noexecstack -I/app/tools/static-openssh/root/include -L. -fPIC -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSLDIR="\"/app/tools/static-openssh/root/ssl\"" -DENGINESDIR="\"/app/tools/static-openssh/root/lib/engines-3\"" -DMODULESDIR="\"/app/tools/static-openssh/root/lib/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG -I/app/tools/static-openssh/root/include -L. -fPIC -MMD -MF crypto/asn1/libcrypto-lib-a_gentm.d.tmp -MT crypto/asn1/libcrypto-lib-a_gentm.o -c -o crypto/asn1/libcrypto-lib-a_gentm.o crypto/asn1/a_gentm.c
550.8 Segmentation fault (core dumped)
550.8 make[1]: *** [Makefile:5069: crypto/asn1/libcrypto-lib-a_gentm.o] Error 139
550.8 make[1]: Leaving directory '/app/tools/static-openssh/build/openssl-3.2.1'
550.8 make: *** [Makefile:2335: build_sw] Error 2
```

### Steps to reproduce

1. First I created a new release
2. Then the dockerhub_sshnpd workflow ran
3. And then it repeatedly failed due to the segfault

### Expected behavior

Images are built successfully

### Additional context

@cconstab remind me why we compile openssh rather than pulling in a package?

The failure seems to be happening in the arm64 build, which I plan to migrate to an arm runner with #1688

--- a/.github/workflows/dockerhub_sshnpd.yml
+++ b/.github/workflows/dockerhub_sshnpd.yml
@@ -10,7 +10,53 @@ permissions: # added using https://github.com/step-security/secure-workflows
   contents: read
 
 jobs:
-  docker:
+  docker_arm64:
+    strategy:
+      matrix:
+        include:
+          - name: sshnpd
+            dockerfile: ./packages/dart/sshnoports/tools/Dockerfile
+          - name: activate_sshnpd
+            dockerfile: ./packages/dart/sshnoports/tools/Dockerfile.activate
+          - name: sshnpd-slim
+            dockerfile: ./packages/dart/sshnoports/tools/Dockerfile.sshnpd-slim
+    runs-on: ubuntu-24.04-arm
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Ensure pubspec.yaml matches git ref (if current git ref is a version tag)
+        shell: bash
+        if: startsWith(github.ref, 'refs/tags/v')
+        working-directory: ./packages/dart/sshnoports
+        run: |
+          REF=${{ github.ref }}
+          VER=${REF:11}
+          sed -i "0,/version:/{s/version: \(.*\)/version: "${VER}"/}" pubspec.yaml
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@53851d14592bedcffcf25ea515637cff71ef929a # v3.3.0
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@6524bf65af31da8d45b59e8c27de4bd072b392f5 # v3.8.0
+      - name: Login to Docker Hub
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      # Extract version for docker tag
+      - name: Get version
+        run: echo "VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
+      - name: Build and push
+        uses: docker/build-push-action@67a2d409c0a876cbe6b11854e3e25193efe4e62d # v6.12.0
+        with:
+          context: .
+          file: ${{ matrix.dockerfile }}
+          platforms: linux/arm64,linux/arm/v7
+          push: true
+          provenance: false
+          tags: |
+            atsigncompany/${{ matrix.name }}:${{ env.VERSION }}
+            atsigncompany/${{ matrix.name }}:release-${{ env.VERSION }}
+
+  docker_amd64:
     strategy:
       matrix:
         include:
@@ -49,14 +95,44 @@ jobs:
         with:
           context: .
           file: ${{ matrix.dockerfile }}
-          platforms: linux/amd64,linux/arm64,linux/arm/v7
+          platforms: linux/amd64
           push: true
           provenance: false
           tags: |
-            atsigncompany/${{ matrix.name }}:${{ env.VERSION }}
-            atsigncompany/${{ matrix.name }}:release-${{ env.VERSION }}
+            atsigncompany/${{ matrix.name }}:amd64-${{ env.VERSION }}
+            atsigncompany/${{ matrix.name }}:amd64-release-${{ env.VERSION }}
+
+  docker_combine:
+    needs: [docker_amd64, docker_arm64]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - name: sshnpd
+          - name: activate_sshnpd
+          - name: sshnpd-slim
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@6524bf65af31da8d45b59e8c27de4bd072b392f5 # v3.8.0
+      - name: Login to Docker Hub
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      # Extract version for docker tag
+      - name: Get version
+        run: echo "VERSION=${GITHUB_REF##*/}" >> $GITHUB_ENV
+      - name: Combine build images
+        id: docker_manifest_build
+        run: |
+          docker buildx imagetools create \
+            -t atsigncompany/${{ matrix.name }}:${{ env.VERSION }} \
+            --append atsigncompany/${{ matrix.name }}:amd64-${{ env.VERSION }}
+          docker buildx imagetools create \
+            -t atsigncompany/${{ matrix.name }}:release-${{ env.VERSION }} \
+            --append atsigncompany/${{ matrix.name }}:amd64-release-${{ env.VERSION }}
       # Promote to latest so long as this isn't a pre-release
-      - name: Tag latest
+      - name: Update latest tag
         run: |
           RELEASE_JSON=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
             "https://api.github.com/repos/${{ github.repository }}/releases/tags/${{ env.VERSION }}")

